Improve link_translate helper

NobodysNightmare opened the pull request:
We now use this helper in more places and make sure it renders primer links that can optionally be localized, if they originate from our static links.

### Ticket

https://community.openproject.org/wp/66029

NobodysNightmare committed:
Use link_translate helper in more places

This is to ensure that we render the same kind of UI in the same way
across the application.

We might soon extend this helper to change rendering to use primer links
or add support for trailing icons on external links.

NobodysNightmare committed:
Use Primer to render links in #link_translate

NobodysNightmare committed:
Use Static::Links.url_for helper more consistently

This is in preparation to extend that helper at a later point to support
localized URLs.

NobodysNightmare commented:
@oliverguenther I think this PR should address all the points from the linked work package, except handling the marking of external links.

As far as I know, we have no central guideline on how to deal with external links so far (e.g. whether they deserve an extra icon or not), so we can add that once we know that.

NobodysNightmare commented on app/forms/work_package_types/subject_configuration_form.rb:
```diff
@@ -84,17 +86,9 @@ def validation_message_for(attribute)
-    end
-
-    def make_link(href, link_text)
-      render(Primer::Beta::Link.new(href:, target: "_blank", underline: true)) { link_text }
```
Note: This was previously rendered with `underline: true`. In the helper I don't specify `underline` at all, but the link is rendered underlined anyways (I believe this is due to changes to the default styling of links).

HDinger replied:
I would like to have `underline` as an option for the `link_translate` method. We already did some logic to underline links in captions (which is why this works here). However, I can imaging other places where we will need to set that manually. We could even think about making it default to `true` because for accessibility reasons it is in most cases better to have it underlined.

NobodysNightmare replied:
Passing underline as an argument to `link_translate` now. It's `true` by default.

oliverguenther commented on config/static_links.yml:
```diff
@@ -38,7 +38,9 @@ docker_installation:
   attribute_highlighting:
-    href: https://www.openproject.org/docs/user-guide/work-packages/work-package-table-configuration/#attribute-highlighting-enterprise-add-on
+    href:
+      de: https://www.openproject.org/de/docs/nutzungshandbuch/arbeitspakete/arbeitspaket-tabellen-konfiguration/#attribute-hervorheben-enterprise-add-on
```
Adding those manually would become error prone and quite tedious, I wonder if @as-op has some ideas on how to extracting metadata about cross-language linking from the website to this YML file, so that devs would only need to maintain the english link perhaps?

NobodysNightmare replied:
True. My current approach is absolutely manual.

It's also noteworthy that the translated pieces of the website do not seem to be part of this repository, so the `check_links` check is also skipping them (making the process more error prone).

as-op replied:
We could implement a "translate/update static links" GitHub Action. The translation for the docs URLs has its single source of truth for the (current) release branch and the dev branch in the Crowdin docs project "[path-fragments.yml](https://crowdin.com/project/openproject-docs/sources/files)".
I can add such an action to my TODO list, but it will take a while before I got the time to do that.

NobodysNightmare replied:
Where does this leave us with this PR? Options I see right now:

* Proceed without translated links (first two commits) for now (because more work is needed
* Proceed with translated links as proposed here, but do not translate a lot of them, because it's a lot of work
* Proceed with translated links, but in a different (yet unsuggested) way

The third option mostly comes from my mind right now, because I am not sure how good of an idea it is to let human and machine edit the same file. Though, for YAML it can usually work, to have both automated and manual edits in the same file.

NobodysNightmare replied:
I've now removed the link translation bits from this PR.

HDinger reviewed (commented):
I just have one minor remark regarding the underline. Could you please also add small section `"Technical notes"` to the [lookbook docs](https://github.com/opf/openproject/blob/dev/lookbook/docs/patterns/04-links.md.erb) about it?
The rest looks fine 👍 🙇

NobodysNightmare committed:
Allow to specify whether links will be underlined

Defaults to true, because underlining them usually helps
with accessibility (making links recognizable as such).

NobodysNightmare committed:
Change target parameter of link_translate to boolean

It's now called external and can only switch between external and non-external links.
This does not only affect the link target, but additionally also adds a trailing visual icon
as it is suggested by our Lookbook.

Since this made me go through all calls of link_translate, I took the opportunity to also replace
occurences of manual calls to OpenProject::Static::Links with the new abbreviated call syntax.

NobodysNightmare commented:
> I just have one minor remark regarding the underline. Could you please also add small section `"Technical notes"` to the [lookbook docs](https://github.com/opf/openproject/blob/dev/lookbook/docs/patterns/04-links.md.erb) about it? The rest looks fine 👍 🙇

Added a small section to the lookbook. The lookbook also reminded me that we were not yet consistently appending a visual icon to external links. I included that in the helper as well, by changing the `target:` parameter to an `external:` parameter.

NobodysNightmare commented on config/locales/en.yml:
```diff
@@ -774,7 +774,7 @@ en:
-          caption_with_supported_attributes_link: Create patterns by adding text, or type "/" to search for %{link}.
+          caption: Create patterns by adding text, or type "/" to search for [supported attributes](attributes_url).
           insert_as_text: "No attributes found. Add as text: \"%{word}\""
           supported_attributes_link: "supported attributes"
```
This translation can be deleted now

NobodysNightmare commented:
Specs are flickering and seem to be unrelated to this PR. Only test failure in last attempt did not fail the attempt before.
